Framework/Core/src/CommonServices.cxx

```diff
@@ -527,22 +527,34 @@ auto sendRelayerMetrics(ServiceRegistry& registry, DataProcessingStats& stats) -
   stats.channelBytesIn.resize(device->fChannels.size());
   stats.channelBytesOut.resize(device->fChannels.size());
   size_t ci = 0;
+  auto lastTotalBytesIn = stats.totalBytesIn.load();
+  auto lastTotalBytesOut = stats.totalBytesOut.load();
```

Comment:
shouldnt't these 2 be `.exchange(0)` instead of `.load()`?
Actually I don't fully understand why these are atomic variables in the first place, but in any case if we use atomics we should use proper atomic semantics. 

At second sight, also below we don't do proper atomic updates ...
I think it should rather be something like this (pseudo-code for only the in-part):
```
uint64_t sum = 0;
for (auto& channel : device->fChannels) sum += channel.second[0].GetBytesRx();
uint64_t lastBytes = stats.totalBytesIn.exchange(sum);
monitoring.send(sum - lastBytes);
```

Reply:
They are atomic because otherwise we used to get a thread-sanitiser warning when running DPL multithreaded albeit I think it cannot actually be called in a multithreaded fashion, so I was just doing something quick. I will update it.